### PR TITLE
Refactor/caller

### PR DESCRIPTION
**Pull request by MayueCif**

### CODE REVIEW

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Mc/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -23,6 +23,23 @@ public static IServiceCollection AddMcClient(this IServiceCollection services, s
         });
     }
 
+    public static IServiceCollection AddMcClient(this IServiceCollection services, Func<Task<string>> mcServiceBaseAddress)
+    {
+        if (mcServiceBaseAddress == null)
+        {
+            throw new ArgumentNullException(nameof(mcServiceBaseAddress));
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Mc/ServiceCollectionExtensions.cs:30]: MasaArgumentException.ThrowIfNull(mcServiceBaseAddress)

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Mc/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -23,6 +23,23 @@ public static IServiceCollection AddMcClient(this IServiceCollection services, s
...
+            {
+                builder.BaseAddress = await mcServiceBaseAddress.Invoke();
+            }, true).AddHttpMessageHandler<HttpClientAuthorizationDelegatingHandler>();
+            callerOptions.DisableAutoRegistration = true;
+        });
+    }
+
     public static IServiceCollection AddMcClient(this IServiceCollection services, Action<CallerOptions> callerOptions)
     {
         ArgumentNullException.ThrowIfNull(callerOptions, nameof(callerOptions));
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Mc/ServiceCollectionExtensions.cs:45]: MasaArgumentException.ThrowIfNull(callerOptions)

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

**Review by zhenlei520 — APPROVED** [review]
LGTM @doddgu

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Service/Caller/Masa.Contrib.Service.Caller.HttpClient/CallerOptionsExtensions.cs [hunk 3]
```diff
@@ -6,33 +6,60 @@ namespace Masa.Contrib.Service.Caller.HttpClient;
 public static class CallerOptionsExtensions
 {
     public static IHttpClientBuilder UseHttpClient(this CallerOptions callerOptions,
-        Func<MasaHttpClientBuilder>? clientBuilder = null)
-        => callerOptions.UseHttpClient(Microsoft.Extensions.Options.Options.DefaultName, clientBuilder);
+        Func<MasaHttpClientBuilder>? clientBuilder = null,
+        bool isSupportUpdate = false)
```
[the remark this answers is not in the document]
Comment (MayueCif) [reply src/Contrib/Service/Caller/Masa.Contrib.Service.Caller.HttpClient/CallerOptionsExtensions.cs:10]: change envirment udpate caller request address

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**